addons/sale/views/variant_templates.xml

```diff
@@ -133,13 +116,21 @@
         </ul>
     </template>
     <template id="badge_extra_price" name="Badge Extra Price">
-        <span class="badge badge-pill badge-light border" t-if="ptav.price_extra">
-        <!-- see note above about price_extra -->
-        <span class="sign_badge_price_extra" t-esc="ptav.price_extra > 0 and '+' or '-'"/>
-        <span t-esc="abs(ptav.price_extra)" class="variant_price_extra text-muted font-italic" style="white-space: nowrap;"
+        <t t-set="combination_info_variant" t-value="product._get_combination_info(ptav, pricelist=pricelist)"/>
+        <span class="badge badge-pill badge-light border" t-if="combination_info_variant['price_extra']">
+        <!--
+            price_extra is displayed as catalog price instead of
+            price after pricelist because it is impossible to
+            compute. Indeed, the pricelist rule might depend on the
+            selected variant, so the price_extra will be different
+            depending on the selected combination. The price of an
+            attribute is therefore variable and it's not very
+            accurate to display it.
```

Comment:
Please update the comment to explain that the price is still converted to the pricelist currency, and that for website scope, price is updated to consider the price-included taxes.